docs/userguide/data.rst

```diff
@@ -3,7 +3,7 @@
 Passing Python objects
 ======================
 
-Parsl apps can communicate via standard Python function parameter passing 
```

Comment:
this PR has a lot of these non-provenance related documentation changes - some of them look like maybe unnecessary reformatting, some are probably worth putting in as their own PR related to documentation cleanup.